cmd/registry.go

```diff
@@ -0,0 +1,136 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ksonnet/ksonnet/metadata"
+	"github.com/ksonnet/ksonnet/utils"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	RootCmd.AddCommand(registryCmd)
+	registryCmd.AddCommand(registryListCmd)
+	registryCmd.AddCommand(registryDescribeCmd)
+}
+
+var registryCmd = &cobra.Command{
+	Use:   "registry",
+	Short: `Manage registries for current project`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return fmt.Errorf("Command 'registry' requires a subcommand\n\n%s", cmd.UsageString())
+	},
+}
+
+var registryListCmd = &cobra.Command{
+	Use:   "list",
+	Short: `List all registries known to current ksonnet app`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		const (
+			nameHeader     = "NAME"
+			protocolHeader = "PROTOCOL"
+			uriHeader      = "URI"
+		)
+
+		if len(args) != 0 {
+			return fmt.Errorf("Command 'registry list' does not take arguments")
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		wd := metadata.AbsPath(cwd)
+
+		manager, err := metadata.Find(wd)
+		if err != nil {
+			return err
+		}
+
+		app, err := manager.AppSpec()
+		if err != nil {
+			return err
+		}
+
+		rows := [][]string{
+			[]string{nameHeader, protocolHeader, uriHeader},
+			[]string{
+				strings.Repeat("=", len(nameHeader)),
+				strings.Repeat("=", len(protocolHeader)),
+				strings.Repeat("=", len(uriHeader)),
+			},
+		}
+		for name, regRef := range app.Registries {
+			rows = append(rows, []string{name, regRef.Protocol, regRef.URI})
+		}
+
+		formatted, err := utils.PadRows(rows)
+		if err != nil {
+			return err
+		}
+		fmt.Print(formatted)
+		return nil
+	},
+	Long: ``,
```

Comment:
Delete the `Long` and `Example` if not going to fill in. Same with Describe below.